### PR TITLE
UNR-2317 Fix crash when starting PIE with missing schema

### PR DESCRIPTION
**Pull request by improbable-valentyn**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
If a class isn't found in schema database, we log an error and try to quit, but if an RPC was getting called, the game would still crash. This will make it early out before hitting that.

#### Primary reviewers
@m-samiec @mattyoung-improbable

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -21,6 +21,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Replicating a static subobject after it has been deleted on a client no longer results in client attaching a new dynamic subobject.
 - Fixed a bug that caused entity pool reservations to cease after a request times out.
 - Running `BuildWorker.bat` for `SimulatedPlayer` no longer fails if the project path has a space in it.
+- The editor no longer crashes if you start PIE with out-of-date schema.
```
Comment (mattyoung-improbable) [inline CHANGELOG.md]:
```suggestion
- Fixed a crash when starting PIE with out-of-date schema.
```

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp [hunk 2]
```diff
@@ -40,6 +40,19 @@ bool USpatialClassInfoManager::TryInit(USpatialNetDriver* InNetDriver, UActorGro
 	return true;
 }
 
+bool USpatialClassInfoManager::CheckClassIsSupported(const FString& PathName)
+{
+	if (!IsSupportedClass(PathName))
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp]: Realise this is just moved code, but we probably shouldn't be doing it in Shipping

File: SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h [hunk 3]
```diff
@@ -92,6 +92,10 @@ class SPATIALGDK_API USpatialClassInfoManager : public UObject
 
 	bool TryInit(USpatialNetDriver* NetDriver, UActorGroupManager* ActorGroupManager);
 
+	// Checks whether a class is supported and quits the game if not. This is to avoid crashing
+	// when running with an out-of-date schema database.
+	bool CheckClassIsSupported(const FString& PathName);
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Public/Interop/SpatialClassInfoManager.h]: This class having a IsSupportedClass, and CheckClassIsSupported functions seems a little obtuse.
Can we rename this to indicate it's intent more clearly?
IsSupportedClass and SafeIsSupportedClass?
IsSupportedClass and ValidateOrExit_IsSupportedClass?

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp [hunk 4]
```diff
@@ -40,13 +40,15 @@ bool USpatialClassInfoManager::TryInit(USpatialNetDriver* InNetDriver, UActorGro
 	return true;
 }
 
-bool USpatialClassInfoManager::CheckClassIsSupported(const FString& PathName)
+bool USpatialClassInfoManager::ValidateOrExit_IsSupportedClass(const FString& PathName)
```
Comment (mattyoung-improbable) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp]: `_` in a method name? Seems bold.

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialClassInfoManager.cpp]: Looks more readable to me

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 5]
```diff
@@ -1183,6 +1183,14 @@ void USpatialNetDriver::ProcessRPC(AActor* Actor, UObject* SubObject, UFunction*
 		GetOrCreateSpatialActorChannel(CallingObject);
 	}
 
+	// If this object's class isn't present in the schema database, we will log an error and tell the
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: This looks a hot path, worth wrapping in SHIPPING tags?

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp]: Discussed offline, will wait to profile before optimising.

**Pull request merged**